hugegraph-mysql/src/main/java/com/baidu/hugegraph/backend/store/mysql/MysqlStore.java

```diff
@@ -168,14 +168,24 @@ public void init() {
     }
 
     @Override
-    public void clear() {
+    public void clear(boolean clearSpace) {
         // Check connected
         this.checkClusterConnected();
 
         if (this.sessions.existsDatabase()) {
-            this.checkOpened();
-            this.clearTables();
-            this.sessions.dropDatabase();
+            if (!clearSpace) {
+                this.checkOpened();
+                this.clearTables();
+                /*
+                 * Disconnect connections for following database drop.
+                 * Connections will be auto reconnected if not drop database
+                 * in next step, but never do this operation because database
+                 * might be blocked in mysql or throw 'terminating' exception.
+                 */
+                this.sessions.resetConnections();
```

Comment:
add more comment: we can't resetConnections() when dropDatabase(), because there are 3 stores(schema,system,graph), which are shared one database, other stores may keep connected with the database when one store doing clear(clearSpace=false).